src/memo/memo.ml

```diff
@@ -553,10 +538,21 @@ module M = struct
     Dep_node
 end
 
+module Considering_state = M.Considering_state
 module State = M.State
 module Dep_node = M.Dep_node
 module Deps = M.Deps
 
+(* For debugging *)
+let _print_dep_node ?prefix (dep_node : _ Dep_node.t) =
+  let prefix =
+    match prefix with
+    | None -> ""
+    | Some prefix -> prefix ^ " "
+  in
+  Format.printf "%s%s\n" prefix
```

Comment:
If we're keeping this function this around, it should probably change to use `Console` to avoid the printout being clobbered by the status line.

Another thing: I think `name_hum` omits the input? That sounds like a pretty confusing printout to me.

Reply:
> If we're keeping this function this around, it should probably change to use `Console` to avoid the printout being clobbered by the status line.

Sounds good, will do.



> Another thing: I think `name_hum` omits the input? That sounds like a pretty confusing printout to me.

I guess you mean I should print out the input here as well (not to change what `name_hum` does). I didn't need inputs to debug the testsuite but sure, I can include it too.

Reply:
I think `name_hum` is a confusing name since the name of the node should include the input. So yeah, I would change what `name_hum` does (or renamed it to e.g. `function_name_hum`).

Reply:
Fixed. I've done it slightly differently, by adding `to_dyn` and printing it out for debugging.